<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
	<arg name="freq" default="400" />

        <node pkg="osrt_ros" type="OnlineUpperLimb" 	name="ik_upperbody_node" output="screen" >
		<!-- param name="resample_period" 		value="0.01" / -->
		<param name="subject_dir" 		value="/mobl2016/" />
		<param name="model_file" 		value="mobl2016_v03.osim" />
		<param name="imu_base_body" 		value="thorax" />
		<rosparam param="imu_observation_order" >['thorax','humerus','radius']</rosparam>
		<param name="imu_direction_axis" 		value="z" />
		<param name="imu_ground_rotation_x" 		value="-90" />
		<param name="imu_ground_rotation_y" 		value="-90" />
		<param name="imu_ground_rotation_z" 		value="0" />
		<param name="rate" 				value="$(arg freq)" />
		<param name="logger_filename_ik" 		value="_ik_upper" />
		<param name="logger_filename_imus" 		value="_imus_upper" />
		<param name="data_save_dir" 		value="/tmp/test" />
		
		<!-- remap from="/agrf_node/input" to="/inverse_kinematics_from_file/r_data"/>
		<remap from="/in_labels" to="/inverse_kinematics_from_file/out_labels"/ -->
		<remap from="/ik_upperbody_node/output" 		to="/ik/output"/>
		<remap from="/ik_upperbody_node/output_filtered" 	to="/ik/output_filtered"/>
		<remap from="/ik_upperbody_node/out_labels" 		to="/ik/outlabels"/>

		<!-- Filter values for filtered output -->
		<param name="filter_output" 		value="true" />
		<param name="memory" 			value="35" />
		<param name="cutoff_freq" 		value="6" />
		<param name="delay" 			value="14" />
		<param name="spline_order" 		value="3" />
        </node>

</launch>

